frmts/vrt/pixelfunctions.cpp

```diff
@@ -479,6 +486,418 @@ static CPLErr ConjPixelFunc(void **papoSources, int nSources, void *pData,
+
+        case GDT_UInt16:
+            return OptimizedSumThroughLargerType<Tsrc, uint16_t>(
+                dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize, nYSize,
+                nSources, papoSources);
+
+        case GDT_Int16:
+            return OptimizedSumThroughLargerType<Tsrc, int16_t>(
+                dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize, nYSize,
+                nSources, papoSources);
+
+        case GDT_Int32:
+            return OptimizedSumThroughLargerType<Tsrc, int32_t>(
+                dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize, nYSize,
+                nSources, papoSources);
+
+        // Float32 and Float64 already covered by OptimizedSum() for packed case
+        default:
+            break;
+    }
+    return false;
+}
+
+/************************************************************************/
+/*                    OptimizedSumThroughLargerType()                   */
+/************************************************************************/
+
+static bool OptimizedSumThroughLargerType(GDALDataType eSrcType,
+                                          GDALDataType eBufType, double dfK,
+                                          void *pOutBuffer, int nPixelSpace,
+                                          int nLineSpace, int nXSize,
+                                          int nYSize, int nSources,
+                                          const void *const *papoSources)
+{
+    switch (eSrcType)
+    {
+        case GDT_Byte:
+            return OptimizedSumThroughLargerType<uint8_t>(
+                eBufType, dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize,
+                nYSize, nSources, papoSources);
+
+        case GDT_UInt16:
+            return OptimizedSumThroughLargerType<uint16_t>(
+                eBufType, dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize,
+                nYSize, nSources, papoSources);
+
+        case GDT_Int16:
+            return OptimizedSumThroughLargerType<int16_t>(
+                eBufType, dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize,
+                nYSize, nSources, papoSources);
+
+        case GDT_Int32:
+            return OptimizedSumThroughLargerType<int32_t>(
+                eBufType, dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize,
+                nYSize, nSources, papoSources);
+
+        case GDT_Float32:
+            return OptimizedSumThroughLargerType<float>(
+                eBufType, dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize,
+                nYSize, nSources, papoSources);
+
+        case GDT_Float64:
+            return OptimizedSumThroughLargerType<double>(
+                eBufType, dfK, pOutBuffer, nPixelSpace, nLineSpace, nXSize,
+                nYSize, nSources, papoSources);
+
+        default:
+            break;
+    }
+
+    return false;
+}
+
+/************************************************************************/
+/*                            SumPixelFunc()                            */
```

Comment:
I have some concerns about how the pixel functions are handling NoData and missing sources. Intrinsics will make this harder to straighten out, but I guess as long as the general case is preserved...